scripts/fetch-github-data.js

```diff
@@ -49,99 +100,101 @@ export const fetchGithubRateLimit = async () => {
+  if (retries < 0) {
+    console.log(`ERROR fetching ${data.githubUrl} - OUT OF RETRIES`);
+    return data;
+  }
+  try {
+    const url = data.githubUrl;
+    const { isMonorepo, repoOwner, repoName, packagePath } = parseUrl(url);
+
+    const result = await makeGraphqlQuery(query, {
+      repoOwner,
+      repoName,
+      packagePath,
+      packageJsonPath: `HEAD:${packagePath === '.' ? '' : `${packagePath}/`}package.json`,
     });
-    let json = await response.json();
-    // console.log(json);
-    result = createRepoDataWithResponse(json);
 
-    if (subrepoData) {
-      result.urls.homepage = subrepoData.homepage;
-      result.name = subrepoData.name;
-      result.topics = subrepoData.topics;
-      result.description = subrepoData.description;
-      result.license = subrepoData.license;
+    if (!result.data.repository && result.errors && result.errors[0].type === 'NOT_FOUND') {
+      const newUrl = await getUpdatedUrl(url);
+      if (newUrl !== url) {
+        console.log(`Repository ${repoOwner}/${repoName} has moved to ${newUrl}`);
+        data.githubUrl = newUrl;
+      } else {
+        console.log(`Repository ${repoOwner}/${repoName} not found`);
+      }
+      return fetchGithubData(data, retries - 1);
     }
 
-    // repoRequestCache[requestUrl] = result;
-    // }
-
+    const github = createRepoDataWithResponse(result.data.repository, isMonorepo);
     return {
       ...data,
-      github: result,
+      github,
     };
   } catch (e) {
     console.log(`retrying fetch for ${data.githubUrl}`);
-    return await fetchGithubData(data);
+    return fetchGithubData(data, retries - 1);
   }
 };
 
-const createRepoDataWithResponse = json => {
+const createRepoDataWithResponse = (json, monorepo) => {
+  if (monorepo && json.packageJson) {
+    const packageJson = JSON.parse(json.packageJson.text);
+    json.homepageUrl = packageJson.homepage;
+    json.name = packageJson.name;
+    json.topics = packageJson.topics;
+    json.description = packageJson.description;
+    json.license = packageJson.license;
+  }
+  const lastCommit = json.defaultBranchRef.target.history.nodes[0].committedDate;
+
   return {
     urls: {
-      repo: json.html_url,
-      clone: json.clone_url,
-      homepage: json.homepage,
+      repo: json.url,
+      clone: `${json.url}.git`,
+      homepage: json.homepageUrl,
     },
     stats: {
-      hasIssues: json.has_issues,
-      hasWiki: json.has_wiki,
-      hasPages: json.has_pages,
-      hasDownloads: json.has_downloads,
-      hasTopics: json.topics ? json.topics.length > 0 : false,
-      updatedAt: json.updated_at,
-      createdAt: json.created_at,
-      pushedAt: json.pushed_at,
-      forks: json.forks,
-      issues: json.open_issues,
-      subscribers: json.subscribers_count,
-      stars: json.stargazers_count,
+      hasIssues: json.hasIssuesEnabled,
+      hasWiki: json.hasWikiEnabled,
+      hasPages: json.deployments.totalCount > 0,
+      hasDownloads: true,
```

Comment:
I couldn't find this in graphql, and I'm not really sure what it is meant to represent. I've set it to `true` to maintain backwards compatibility as every repository listed in the data was `true`

Reply:
I think it probably refers to the repo having files/installers attached to releases, but not sure tbh.